src/pages/Swap/Swap.tsx

```diff
@@ -176,161 +91,201 @@ const toToFrom = (
     : undefined;
 };
 
-const isFromFieldAssetChanged = (
-  value: SwapFormModel,
-  prevValue: SwapFormModel,
-): boolean => value?.from?.asset?.id !== prevValue?.from?.asset?.id;
+const getToAssets = (fromAsset?: string) =>
+  fromAsset ? getAvailableAssetFor(fromAsset) : assets$;
+
+const getSelectedPool = (
+  xId?: string,
+  yId?: string,
+): Observable<AmmPool | undefined> =>
+  xId && yId
+    ? getPoolByPair(xId, yId).pipe(
+        map((pools) => maxBy(pools, (p) => p.lp.amount)),
+      )
+    : of(undefined);
+
+export const Swap = () => {
+  const form = useForm<SwapFormModel>({
+    fromAmount: undefined,
+    toAmount: undefined,
+    fromAsset: {
+      name: 'ERG',
+      id: '0000000000000000000000000000000000000000000000000000000000000000',
+      decimals: ERG_DECIMALS,
+    },
+    toAsset: undefined,
+    pool: undefined,
+  });
+  const [fromAssets] = useObservable(assets$);
+  const [toAssets, updateToAssets] = useSubject(getToAssets);
+  const [balance] = useWalletBalance();
+  const [{ minerFee }] = useSettings();
 
-const isToAssetChanged = (
-  value: SwapFormModel,
-  prevValue: SwapFormModel,
-): boolean =>
-  !!value?.from?.asset &&
-  !!value?.to?.asset &&
-  value?.to?.asset?.id !== prevValue?.to?.asset?.id;
+  const getInsufficientTokenForFee = useCallback(
+    (value: SwapFormModel) => {
+      const { fromAmount, fromAsset } = value;
+      let totalFees = +calculateTotalFee(
+        [minerFee, UI_FEE, defaultExFee],
+        ERG_DECIMALS,
+      );
+      totalFees =
+        fromAsset?.id === ERG_TOKEN_ID
+          ? totalFees + fromAmount?.value!
+          : totalFees;
+
+      return +totalFees > balance.get(ERG_TOKEN_ID)
+        ? ERG_TOKEN_NAME
+        : undefined;
+    },
+    [minerFee, balance],
+  );
 
-const getAvailablePools = (xId?: string, yId?: string): Observable<AmmPool[]> =>
-  xId && yId ? getPoolByPair(xId, yId) : of([]);
+  const getInsufficientTokenForTx = useCallback(
```

Comment:
I guess this function returns a string with a token name. So let's rename it to `...TokenName...`